### ISSUE TITLE
Result of cyclic-import check [R0401] depends on execution speed

### ISSUE THREAD

**Issue by jpieger** [PyCQA/pylint#7137]
### Bug description

There seems to be a race condition in the cyclic-import checker. When pylint is executed with multiple jobs, then it can miss cyclic imports on fast machines.

File `example/a.py`:
```python
# pylint: disable=unused-import,missing-module-docstring
import b
```
File `example/b.py`
```python
# pylint: disable=unused-import,missing-module-docstring
import a
```

### Configuration

```ini
[MAIN]

# With this configuration the issue is found
jobs=1

# With this configuration it is missed
# jobs=2
```

### Command used

```shell
pylint example/
```

### Pylint output

```shell
Running with jobs=1:

$ pylint example
************* Module a
example/a.py:1:0: R0401: Cyclic import (a -> b) (cyclic-import)
------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)

Running with jobs=2:
$ pylint example

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 5.00/10, +5.00)
```

### Expected behavior

Changing the number of jobs should not affect the result.

### Pylint version

```shell
pylint 2.14.4
astroid 2.11.6
Python 3.9.13 (main, May 18 2022, 19:39:14) 
[GCC 12.1.0]
```

### OS / Environment

Problem was observed on Arch Linux, Ubuntu 20.04 and Windows 10

**Comment by jacobtylerwalls** [comment]
Duplicate of #374
